### ISSUE TITLE
HELP: Product View doesn't show any data

### ISSUE THREAD

**Issue by karlpanganiban** [Hygieia/Hygieia#3291]
I already finished setting up all widgets. Here's what it looks like. (Feature / Deploy / Build / Repo)
![image](https://user-images.githubusercontent.com/14307099/89254915-ecfeb800-d652-11ea-81f3-9961c4cd6e34.png)

However, my PRODUCT VIEW doesn't show anything. What am I missing? 
![image](https://user-images.githubusercontent.com/14307099/89254991-13bcee80-d653-11ea-9862-3b5fed0d4d8e.png)

Thanks in advance.

**Comment by larsen-mallick** [comment]
The problem was a code error. PipelineController.java line  33 is incorrect.

GET Method does not accept @RequestBody. I made the following update and it works -

@RequestMapping(value = "/pipeline", method = GET, produces = APPLICATION_JSON_VALUE)
public Iterable<PipelineResponse> searchPipelines(@RequestParam String collectorItemId, @RequestParam String beginDate,
                                                  @RequestParam String endDate) throws HygieiaException {
    PipelineSearchRequest pipelineSearchRequest = new PipelineSearchRequest();
    pipelineSearchRequest.setBeginDate(Long.valueOf(beginDate));
    pipelineSearchRequest.setEndDate(Long.valueOf(endDate));
    List<ObjectId> collectorItemIdList = new ArrayList<>();
    collectorItemIdList.add(new ObjectId(collectorItemId));
    pipelineSearchRequest.setCollectorItemId(collectorItemIdList);
    return pipelineService.search(pipelineSearchRequest);
}

**Comment by karlpanganiban** [comment]
@larsen-mallick I tried what you suggested, but my build FAILED already.
![image](https://user-images.githubusercontent.com/14307099/89533618-466b1080-d826-11ea-9f04-07ad289803ac.png)

Here's what my code looked like after doing what you suggested (Edited file: **PipelinController.java** inside api repo)
```
package com.capitalone.dashboard.rest;

import com.capitalone.dashboard.misc.HygieiaException;
import com.capitalone.dashboard.model.PipelineResponse;
import com.capitalone.dashboard.request.PipelineSearchRequest;
import com.capitalone.dashboard.service.PipelineService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;

import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
import static org.springframework.web.bind.annotation.RequestMethod.GET;

@RestController
public class PipelineController {
    private final PipelineService pipelineService;

    @Autowired
    public PipelineController(PipelineService pipelineService) {
        this.pipelineService = pipelineService;
    }

    // THIS IS THE OLD CODE
    //@RequestMapping(value = "/pipeline", method = GET, produces = APPLICATION_JSON_VALUE)
    //public Iterable<PipelineResponse> searchPipelines(@Valid @RequestBody PipelineSearchRequest searchRequest) throws HygieiaException {
    //    return pipelineService.search(searchRequest);
    //}

    @RequestMapping(value = "/pipeline", method = GET, produces = APPLICATION_JSON_VALUE)
    public Iterable searchPipelines(@RequestParam String collectorItemId, @RequestParam String beginDate, @RequestParam String endDate) throws HygieiaException {
        PipelineSearchRequest pipelineSearchRequest = new PipelineSearchRequest();
        pipelineSearchRequest.setBeginDate(Long.valueOf(beginDate));
        pipelineSearchRequest.setEndDate(Long.valueOf(endDate));
        List collectorItemIdList = new ArrayList<>();
        collectorItemIdList.add(new ObjectId(collectorItemId));
        pipelineSearchRequest.setCollectorItemId(collectorItemIdList);
        return pipelineService.search(pipelineSearchRequest);
    }
}
```
What am I missing? If this is a code issue. So everyone has this bug.?

**Comment by larsen-mallick** [comment]
Looks like the following import statements are missing. 

import org.springframework.web.bind.annotation.RequestParam;

import java.util.ArrayList;
import java.util.List;

import org.bson.types.ObjectId;

**Comment by karlpanganiban** [comment]
@larsen-mallick thanks, would you mind sharing me your **PipelinController.java** file? It's not that very long. just to make sure. :)

**Comment by karlpanganiban** [comment]
@larsen-mallick I already added the import missing statements your provide and build was successful, but still no luck in my Product Page. Still no data showing, same with the picture I showed above.

In the good side, I don't anymore see my previous error in API logs when I try to access my Product Page, like this.

`2020-08-07 09:54:34,235 [http-nio-8080-exec-4] ERROR c.c.d.rest.RestApiExceptionHandler - Required request body is missing: public java.lang.Iterable<com.capitalone.dashboard.model.PipelineResponse> com.capitalone.dashboard.rest.PipelineController.searchPipelines(com.capitalone.dashboard.request.PipelineSearchRequest) throws com.capitalone.dashboard.misc.HygieiaException`

In short, **no more api errors log now, but still no data.** Anything I need to adjust too?

**Comment by larsen-mallick** [comment]
Do you have it as POST method or a GET method?

**Comment by karlpanganiban** [comment]
@larsen-mallick Where do I have to paste that method? Here's my current code

- Still no data in product view
- No more error in the API logs
```
package com.capitalone.dashboard.rest;

import com.capitalone.dashboard.misc.HygieiaException;
import com.capitalone.dashboard.model.PipelineResponse;
import com.capitalone.dashboard.request.PipelineSearchRequest;
import com.capitalone.dashboard.service.PipelineService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;

// these 3 are test only
import org.springframework.web.bind.annotation.RequestParam;
import java.util.ArrayList;
import java.util.List;
import org.bson.types.ObjectId;

import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
import static org.springframework.web.bind.annotation.RequestMethod.GET;

@RestController
public class PipelineController {
    private final PipelineService pipelineService;

    @Autowired
    public PipelineController(PipelineService pipelineService) {
        this.pipelineService = pipelineService;
    }

    public Iterable searchPipelines(@RequestParam String collectorItemId, @RequestParam String beginDate, @RequestParam String endDate) throws HygieiaException {
        PipelineSearchRequest pipelineSearchRequest = new PipelineSearchRequest();
        pipelineSearchRequest.setBeginDate(Long.valueOf(beginDate));
        pipelineSearchRequest.setEndDate(Long.valueOf(endDate));
        List collectorItemIdList = new ArrayList<>();
        collectorItemIdList.add(new ObjectId(collectorItemId));
        pipelineSearchRequest.setCollectorItemId(collectorItemIdList);
        return pipelineService.search(pipelineSearchRequest);
    } 
}
```
Would you mind just sharing me your **PipelineController.java** file?

**Comment by karlpanganiban** [comment]
@larsen-mallick it's working now.. I checked and didn't noticed I included the GET method when I commented out the old code
` @RequestMapping(value = "/pipeline", method = GET, produces = APPLICATION_JSON_VALUE)`

Thanks a lot for the help! :) Btw, do you know why I have a gap in the right side part of the dashboard? Is ti because I only have 3 stages (DEV/STG/PRD)?
![image](https://user-images.githubusercontent.com/14307099/89732845-06a16480-da84-11ea-957f-41c278d953f2.png)

**Comment by karlpanganiban** [comment]
@larsen-mallick also, do you happen to know why I don't have the GREEN and RED boxes on each column like this one from the hygieia documentation pag.
![image](https://user-images.githubusercontent.com/14307099/89733117-de1a6a00-da85-11ea-80d5-6c74395fcd57.png)

**Comment by larsen-mallick** [comment]
Sorry, am also trying to setup the entire product dashboard. 

Would you be able to share how you have setup the DEV/STG/PROD data on product dashboard?

**Comment by karlpanganiban** [comment]
@larsen-mallick the DEV/STG/PRD is pulled in the Deploy Widget. So you must have a working Deploy widget first
![image](https://user-images.githubusercontent.com/14307099/89753342-7ce8aa00-db0a-11ea-9744-ce3c9041a541.png)

Here's the code block in Jenkinsfile that pushes result in deploy widget ( I am using [jenkins publisher](https://github.com/Hygieia/hygieia-publisher-jenkins-plugin))
```
                hygieiaArtifactPublishStep artifactDirectory: '/target', artifactGroup: 'com.capitalone.dashboard', artifactName: '*.jar', artifactVersion: '1.0'
                hygieiaDeployPublishStep applicationName: 'hygieia', artifactDirectory: '/target', artifactGroup: 'com.capitalone.dashboard', artifactName: '*.jar', artifactVersion: '1.0', buildStatus: 'Success', environmentName: 'Dev'
```

**Comment by syedsfayaz** [comment]
@karlpanganiban I have made the changes as you did but I dont see any difference in my product dashboard. Can you please tell me what am I missing here?

```
package com.capitalone.dashboard.rest;

import com.capitalone.dashboard.misc.HygieiaException;
import com.capitalone.dashboard.model.PipelineResponse;
import com.capitalone.dashboard.request.PipelineSearchRequest;
import com.capitalone.dashboard.service.PipelineService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;

import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
import static org.springframework.web.bind.annotation.RequestMethod.GET;

import org.springframework.web.bind.annotation.RequestParam;

import java.util.ArrayList;
import java.util.List;

import org.bson.types.ObjectId;

@RestController
public class PipelineController {
    private final PipelineService pipelineService;

    @Autowired
    public PipelineController(PipelineService pipelineService) {
        this.pipelineService = pipelineService;
    }

    //@RequestMapping(value = "/pipeline", method = GET, produces = APPLICATION_JSON_VALUE)
    //public Iterable<PipelineResponse> searchPipelines(@Valid @RequestBody //PipelineSearchRequest searchRequest) throws HygieiaException {
    //    return pipelineService.search(searchRequest);
    //}
	//@RequestMapping(value = "/pipeline", method = GET, produces = APPLICATION_JSON_VALUE)
    public Iterable searchPipelines(@RequestParam String collectorItemId, @RequestParam String beginDate, @RequestParam String endDate) throws HygieiaException {
        PipelineSearchRequest pipelineSearchRequest = new PipelineSearchRequest();
        pipelineSearchRequest.setBeginDate(Long.valueOf(beginDate));
        pipelineSearchRequest.setEndDate(Long.valueOf(endDate));
        List collectorItemIdList = new ArrayList<>();
        collectorItemIdList.add(new ObjectId(collectorItemId));
        pipelineSearchRequest.setCollectorItemId(collectorItemIdList);
        return pipelineService.search(pipelineSearchRequest);
    }
}
```

**Comment by karlpanganiban** [comment]
@syedsfayaz you commented out the @RequestMapping line.. it should be a part of the code. Btw, here's my PipelineController.java

```
package com.capitalone.dashboard.rest;

import com.capitalone.dashboard.misc.HygieiaException;
import com.capitalone.dashboard.model.PipelineResponse;
import com.capitalone.dashboard.request.PipelineSearchRequest;
import com.capitalone.dashboard.service.PipelineService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;

import org.springframework.web.bind.annotation.RequestParam;
import java.util.ArrayList;
import java.util.List;
import org.bson.types.ObjectId;

import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
import static org.springframework.web.bind.annotation.RequestMethod.GET;

@RestController
public class PipelineController {
    private final PipelineService pipelineService;

    @Autowired
    public PipelineController(PipelineService pipelineService) {
        this.pipelineService = pipelineService;
    }

    @RequestMapping(value = "/pipeline", method = GET, produces = APPLICATION_JSON_VALUE)

    public Iterable searchPipelines(@RequestParam String collectorItemId, @RequestParam String beginDate, @RequestParam String endDate) throws HygieiaException {
        PipelineSearchRequest pipelineSearchRequest = new PipelineSearchRequest();
        pipelineSearchRequest.setBeginDate(Long.valueOf(beginDate));
        pipelineSearchRequest.setEndDate(Long.valueOf(endDate));
        List collectorItemIdList = new ArrayList<>();
        collectorItemIdList.add(new ObjectId(collectorItemId));
        pipelineSearchRequest.setCollectorItemId(collectorItemIdList);
        return pipelineService.search(pipelineSearchRequest);
    } 
}
```

**Comment by rvema** [comment]
This issue should have been fixed as part of https://github.com/Hygieia/api/pull/163 can you retry?

**Comment by syedsfayaz** [comment]
@karlpanganiban Thanks for sharing the code snippet. I am able to see some data in the product dashboard. But I dont see any commits or other data. Were you able to resolve this issue. I did a test commit and ran build and deploy job but no results are displayed. 
![image](https://user-images.githubusercontent.com/26099918/92623777-8f761080-f27b-11ea-922e-400c7898c6f6.png)

Here is my TeamDashboard
![image](https://user-images.githubusercontent.com/26099918/92623837-a74d9480-f27b-11ea-8381-948c76b79bea.png)

**Comment by shravan-puttur** [comment]
> @larsen-mallick also, do you happen to know why I don't have the GREEN and RED boxes on each column like this one from the hygieia documentation pag.
> ![image](https://user-images.githubusercontent.com/14307099/89733117-de1a6a00-da85-11ea-80d5-6c74395fcd57.png)

Hi @larsen-mallick , Do we need Hygieia-Jenkins Plugin configured in our Jenkins in order for it to display data in product dashboard? We are running github collector and build, deploy collectors and configured the team dashboard. But the product dashboard does not show any data.

**Comment by syedsfayaz** [comment]
@shravan-puttur   Can you please share what exactly you are seeing on your product dashboard? I am having similar issue and I am using jenkins plugin to push data to Hygieia API.

**Comment by avinashkukutla** [comment]
> hygieiaArtifactPublishStep

@karlpanganiban @larsen-mallick  We have followed instruction as you guys specified . But still we are not able to display deploy details in product dashboard .I got below Hygieia version from pom.xml

Hygieia API -- 3.1.2-SNAPSHOT

Hygieia UI --- 3.1.1-SNAPSHOT

Can some one help me to resolve the issue.
